### PR TITLE
Adding iterator method to Dict

### PR DESCRIPTION
**Pull request by kellrott**
Addresses #58

### CODE REVIEW

File: py/dict.go [hunk 1]
```diff
@@ -100,6 +100,16 @@ func (a StringDict) M__repr__() (Object, error) {
 	return String(out.String()), nil
 }
 
+
+func (d StringDict) M__iter__() (Object, error) {
```
Comment (sbinet) [inline py/dict.go]: please run `gofmt -w` on this file.

**Comment by ncw** [comment]
According to the coverage report your test doesn't hit your added code?

https://codecov.io/gh/go-python/gpython/pull/59/diff

**Comment by kellrott** [comment]
The code is covered in `py/tests/dict.py` line 17, but the system doesn't seem to recognize that.

File: py/tests/dict.py [hunk 2]
```diff
@@ -12,4 +12,8 @@
 a = repr({"a":"b","c":5.5})
 assert a == "{'a': 'b', 'c': 5.5}" or a == "{'c': 5.5, 'a': 'b'}"
 
+doc="check __iter__"
+a = {"a":"b","c":5.5}
+assert "a" in a
```
Comment (ncw) [inline py/tests/dict.py]: Strictly speaking this test will no longer work when we implement the `__contains__` method for dict.

Do can you change it to something like

```python
l =  list(iter(a))
assert "a" in l
assert "c" in l
assert len(l) == 2
```

Comment (kellrott) [reply py/tests/dict.py]: This should be fixed now

**Comment by ncw** [comment]
That looks great now thank you!  I see you added an items method too :-)

Will merge now.

**Pull request merged**